### PR TITLE
Run the C# tests on mingw too

### PR DESCRIPTION
**Pull request by dra27**
While working on the second part of #9939, I had need to run the `lib-dynlink-csharp` tests on mingw, rather than msvc. This PR enables that - all that's needed is the addition of `${libext}` in ocamltest. The motion in the AppVeyor script is to ensure that a Microsoft build environment is always available so that the mingw32 test has access to `csc`.

### CODE REVIEW

**Comment by dra27** [comment]
Turns out we should have been doing this already, as this hits the issue even with mingw64-runtime 7.0.0! This PR will succeed when #10046 is merged.

File: Changes
Comment (gasche) [inline Changes]: this should be filled before merging

**Review by gasche — APPROVED** [review]
Looks good to me.

**Comment by gasche** [comment]
AppVeyor appears to fail for mingw32.

**Comment by dra27** [comment]
~Blocked on #10357~

**Pull request merged**
